modules/renter/repair.go

```diff
@@ -253,14 +268,31 @@ func (r *Renter) managedRepairIteration(rs *repairState) {
 	var chunksToDelete []chunkID
 	for chunkID, chunkStatus := range rs.incompleteChunks {
 		// check if the chunk is currently being downloaded for recovery
-		if _, downloading := rs.downloadingChunks[chunkID]; downloading {
-			continue
+		dc, downloading := rs.downloadingChunks[chunkID]
+		if downloading {
+			// Check if download finished
+			select {
+			default:
+				// Check if download timed out
+				if time.Now().After(dc.startTime.Add(chunkDownloadTimeout)) {
+					r.log.Println("Download of chunk to repair timed out")
+					chunksToDelete = append(chunksToDelete, chunkID)
+					delete(rs.downloadingChunks, chunkID)
+				}
+				continue
+			case <-dc.d.downloadFinished:
+				// Download finished, continue repairing the chunk.
+			}
 		}
-		// Update the number of gaps for this chunk.
+
+		// Don't update gaps for downloaded chunks twice
```

Comment:
Where do the download gaps get updated? It's not a big deal if the code is working, because we're scrapping it anyway, but generally speaking we should have everything happening all in one place so that it's easier to keep track of in the future.

But again, this code is all going into a dumpster in a few weeks anyway, we've got a much better way of doing this. So, we can relax a bit here and just be happy that it's not buggy.

Reply:
Right below. They are still updated but the comment was moved inside an if statement.